Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/interfaces.go

```diff
@@ -52,6 +95,20 @@ type ObjectCreater interface {
 	New(version, kind string) (out Object, err error)
 }
 
+// ObjectDecoder is a convenience interface for identifying serialized versions of objects
+// and transforming them into Objects. It intentionally overlaps with ObjectTyper and
+// Decoder for use in decode only paths.
+type ObjectDecoder interface {
```

Comment:
Decoder is the minimal interface.  ObjectDecoder is the interface for a scheme.

----- Original Message -----
> > @@ -52,6 +95,20 @@ type ObjectCreater interface {
> >  	New(version, kind string) (out Object, err error)
> >  }
> >  
> > +// ObjectDecoder is a convenience interface for identifying serialized
> > versions of objects
> > +// and transforming them into Objects. It intentionally overlaps with
> > ObjectTyper and
> > +// Decoder for use in decode only paths.
> > +type ObjectDecoder interface {
> 
> Why shouldn't these methods be on the Decoder interface itself?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1959/files#r29329435